scene/gui/color_picker.cpp

```diff
@@ -93,6 +93,30 @@ void ColorPicker::set_focus_on_line_edit() {
 
 void ColorPicker::_update_controls() {
 
+	const char *rgb[3] = { "R", "G", "B" };
+	const char *hsv[3] = { "H", "S", "V" };
+
+	for (int i = 0; i < 3; i++) {
+		if (hsv_mode_enabled)
+			labels[i]->set_text(hsv[i]);
+		else
+			labels[i]->set_text(rgb[i]);
+	}
+
+	if (hsv_mode_enabled) {
+		set_raw_mode(false);
+		btn_raw->set_disabled(true);
+	} else {
+		btn_raw->set_disabled(false);
+	}
+
+	if (raw_mode_enabled) {
```

Comment:
Is optimizing a single check worth making the code look like this?
```
	if (hsv_mode_enabled) {
		set_raw_mode(false);
		btn_raw->set_disabled(true);
	} else {
		btn_raw->set_disabled(false);

		if (raw_mode_enabled) {
			set_hsv_mode(false);
			btn_hsv->set_disabled(true);
		} else {
			btn_hsv->set_disabled(false);
		}
	}
```
No need to move `btn_hsv->set_disabled(false);`, because it's impossible to disable raw mode and enable hsv mode at the same time.

Reply:
Another option is to merge the two if blocks into a single check somehow,.

```
if (!hsv_mode_enabled && !raw_mode_enabled) {
	btn_raw->set_disabled(false);
	btn_hsv->set_disabled(false);
} else if (!hsv_mode_enabled && raw_mode_enabled) {
	btn_raw->set_disabled(true);
	set_hsv_mode(false);
	btn_hsv->set_disabled(true);
} else {
	set_raw_mode(false);
	btn_raw->set_disabled(true);
}
```

Reply:
@gdupr That would actually increase the average number of checks.

Reply:
@aaronfranke, We can change it to make it work with the same amount of checks as suggested. The suggestion looks like this:

1. if (hsv_mode_enabled)
2. if (hsv_mode_enabled) + if (raw_mode_enabled)

With the suggestion below you have this pattern(the first 2. check is false):
1. if (hsv_mode_enabled)
2. if (hsv_mode_enabled) + if (!raw_mode_enabled)

```
if (hsv_mode_enabled) {
	set_raw_mode(false);
	btn_raw->set_disabled(true);
} else if (!raw_mode_enabled) {
	btn_raw->set_disabled(false);
	btn_hsv->set_disabled(false);
} else { // if !hsv_mode_enabled && raw_mode_enabled
	btn_raw->set_disabled(true);
	set_hsv_mode(false);
	btn_hsv->set_disabled(true);
}
```

We can also make the same check-pattern.

```
if (hsv_mode_enabled) {
	set_raw_mode(false);
	btn_raw->set_disabled(true);
} else if (raw_mode_enabled) {
	btn_raw->set_disabled(true);
	set_hsv_mode(false);
	btn_hsv->set_disabled(true);
} else { // if !hsv_mode_enabled && !raw_mode_enabled
	btn_raw->set_disabled(false);
	btn_hsv->set_disabled(false);
}
```.

Reply:
The last one actually makes sense. Changed. I also optimized the for loop above.